core/dbt/context/providers.py

```diff
@@ -554,8 +554,11 @@ def create_relation(self, target_model: ManifestNode) -> RelationProxy:
             and target_model.defer_relation
             and self.config.args.defer
             and (
-                # User has explicitly opted to prefer defer_relation
-                self.config.args.favor_state
+                # User has explicitly opted to prefer defer_relation for unselected resources
+                (
+                    self.config.args.favor_state
+                    and target_model.unique_id not in selected_resources.SELECTED_RESOURCES
```

Comment:
This is a global (ew), and has been for a while:
- https://github.com/dbt-labs/dbt-core/issues/6009